### PR TITLE
Fix mongo version checks

### PR DESCRIPTION
**Pull request by metamben**

### CODE REVIEW

File: modules/drivers/mongo/src/metabase/driver/mongo.clj [hunk 1]
```diff
@@ -228,30 +229,24 @@
                  :standard-deviation-aggregations]]
   (defmethod driver/supports? [:mongo feature] [_driver _feature] true))
 
-(defn- db-version [db]
-  (get-in db [:details :version]))
+(defmethod driver/database-supports? [:mongo :expressions]
+  [_driver _feature db]
+  (boolean (some-> (:dbms_version db)
+                   :semantic-version
+                   (driver.u/semantic-version-gte [4]))))
```
Comment (metamben) [inline modules/drivers/mongo/src/metabase/driver/mongo.clj:236]: The DB must have been synced.

File: test/metabase/driver/util_test.clj [hunk 2]
```diff
@@ -239,15 +239,17 @@
...
     (is (true? (driver.u/semantic-version-gte [4 nil] [4])))
     (is (false? (driver.u/semantic-version-gte [3] [4])))
     (is (false? (driver.u/semantic-version-gte [3] [4 nil])))
     (is (false? (driver.u/semantic-version-gte [4] [4 1])))
     (is (false? (driver.u/semantic-version-gte [4 nil] [4 1])))
+    (is (false? (driver.u/semantic-version-gte [4 0] [4 0 1])))
     (is (false? (driver.u/semantic-version-gte [4 0 1] [4 1])))
-    (is (false? (driver.u/semantic-version-gte [3 9] [4 0])))))
+    (is (false? (driver.u/semantic-version-gte [3 9] [4 0])))
+    (is (false? (driver.u/semantic-version-gte [3 1] [4])))))
```
Comment (metamben) [inline test/metabase/driver/util_test.clj:255]: The original version returned `true` in this case.

**Review by snoe — APPROVED** [review]
LGTM

**Comment by metamben** [comment]
@calherries there are a bunch of test failing because expressions are now supported. Should I fix them or have you already worked on these?

**Comment by calherries** [comment]
@metamben I haven't worked on any of the failures. Only the datetimeDiff and now function.

File: test/metabase/driver/util_test.clj (hunk 2 above)
Comment (calherries) [reply test/metabase/driver/util_test.clj:255]: Perhaps we should tests for `nil` arguments? Maybe this should be an assertion and throw an error?

**Review by calherries — APPROVED** [review]
Looks like a good improvement over my attempt at fixing it!

I still wonder if it could explicitly handle `nil` better though by throwing an error. I don't think `:semantic-version` will exist for every DBMS so there's a decent chance we could accidentally introduce a bug by giving it a `nil` argument which would be hard to track.

**Comment by metamben** [comment]
Closing in favor of #27176.

**Pull request closed without merging**